clang/lib/AST/ODRHash.cpp

```diff
@@ -709,6 +712,11 @@ void ODRHash::AddFunctionDecl(const FunctionDecl *Function,
     return;
   }
 
+  if (DeletedMessage) {
+    ID.AddString(DeletedMessage->getBytes());
+    return;
+  }
+
```

Comment:
I would move that after `AddBoolean(DeletedMessage);` - it make no sense to separate the delete specifier from its message. (there is a question of whether isExplicitlyDefaulted/isDeletedAsWritten should be after skip body, but that seems  logical as is, and at worse it would be pre-existing)

Reply:
> (there is a question of whether isExplicitlyDefaulted/isDeletedAsWritten should be after skip body, but that seems logical as is, and at worse it would be pre-existing)

Yeah, that’s true; considering those are already always included it makes sense to always include the message as well if it exists.